### PR TITLE
[master] Cherrypicking 3.3.5a -> master

### PR DESCRIPTION
**Pull request by funjoker**
**Changes proposed:**

-  Some 

**Target branch(es):** 3.3.5/master

- [ ] 3.3.5
- [x] master

**Issues addressed:** Closes #  (insert issue tracker number)

**Tests performed:** (Does it build, tested in-game, etc.)

**Known issues and TODO list:** (add/remove lines as needed)

- [ ] 
- [ ]

### CODE REVIEW

**Comment by funjoker** [comment]
no more updates to this

File: src/server/game/Entities/Creature/Creature.cpp [hunk 1]
```diff
@@ -1891,29 +1934,23 @@ void Creature::ForcedDespawn(uint32 timeMSToDespawn, Seconds const& forceRespawn
...
-            m_corpseDelay = 0;
-            setDeathState(JUST_DIED);
-            m_respawnDelay = respawnDelay;
-            m_corpseDelay = corpseDelay;
-        }
-        else
+        setDeathState(JUST_DIED);
+
+        if (forceRespawnTimer > Seconds::zero())
         {
```
Comment (funjoker) [inline src/server/game/Entities/Creature/Creature.cpp]: ForcedDespawn has been far different in master than in 335...
We should check if this i now ok

**Comment by msoky** [comment]
ready to merge after a month?

**Comment by Shauren** [comment]
Had to fix a few Set**Flag -> Add**Flag

**Comment by funjoker** [comment]
Ah thanks

**Pull request merged**